config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/builder/JmsDestinationEntityBuilder.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright (c) 2018 CA. All rights reserved.
+ * This software may be modified and distributed under the terms
+ * of the MIT license.  See the LICENSE file for details.
+ */
+
+package com.ca.apim.gateway.cagatewayconfig.bundle.builder;
+
+import com.ca.apim.gateway.cagatewayconfig.beans.Bundle;
+import com.ca.apim.gateway.cagatewayconfig.beans.JmsDestination;
+import com.ca.apim.gateway.cagatewayconfig.util.IdGenerator;
+import com.ca.apim.gateway.cagatewayconfig.util.entity.EntityTypes;
+import com.google.common.collect.ImmutableMap;
+import org.jetbrains.annotations.NotNull;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static com.ca.apim.gateway.cagatewayconfig.util.entity.EntityTypes.CASSANDRA_CONNECTION_TYPE;
+import static com.ca.apim.gateway.cagatewayconfig.util.gateway.BuilderUtils.buildAndAppendPropertiesElement;
+import static com.ca.apim.gateway.cagatewayconfig.util.gateway.BundleElementNames.*;
+import static com.ca.apim.gateway.cagatewayconfig.util.properties.PropertyConstants.*;
+import static com.ca.apim.gateway.cagatewayconfig.util.xml.DocumentUtils.createElementWithAttributesAndChildren;
+import static com.ca.apim.gateway.cagatewayconfig.util.xml.DocumentUtils.createElementWithTextContent;
+
+@Singleton
+public class JmsDestinationEntityBuilder implements EntityBuilder {
+    private static final Integer ORDER = 1500;
+    
+    private final IdGenerator idGenerator;
+    
+    @Inject
+    JmsDestinationEntityBuilder(IdGenerator idGenerator) {
+        this.idGenerator = idGenerator;
+    }
+    
+    @Override
+    public List<Entity> build(Bundle bundle, BundleType bundleType, Document document) {
+        switch (bundleType) {
+            case DEPLOYMENT:
+                return bundle.getJmsDestinations().entrySet().stream()
+                        .map(e -> EntityBuilderHelper.getEntityWithOnlyMapping(CASSANDRA_CONNECTION_TYPE, e.getKey(), idGenerator.generate()))
+                        .collect(Collectors.toList());
+            case ENVIRONMENT:
+                return bundle.getJmsDestinations().entrySet().stream().map(e ->
+                        buildEntity(bundle, e.getKey(), e.getValue(), document)
+                ).collect(Collectors.toList());
+            default:
+                throw new EntityBuilderException("Unknown bundle type: " + bundleType);
+        }
+    }
+
+    private Entity buildEntity(Bundle bundle, String name, JmsDestination jmsDestination, Document document) {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Remove this unused method parameter "bundle". ([squid:S1172](https://sonarcloud.io/organizations/ca-api-gateway/rules#rule_key=squid%3AS1172))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=com.ca.apim.gateway%3Agateway-developer-plugin&pullRequest=138&issues=AWc81LiZfimTzgTvo62B&open=AWc81LiZfimTzgTvo62B)

[ISSUE_KEY]: AWc81LiZfimTzgTvo62B